detection_rules/rule.py

```diff
@@ -316,7 +341,42 @@ def dict_hash(cls, contents, versioned=True):
 
     def get_hash(self):
         """Get a standardized hash of a rule to consistently check for changes."""
-        return self.dict_hash(self.contents)
+        return self.dict_hash(self.get_payload())
+
+    def get_version(self):
+        """Get the version of the rule."""
+        from .packaging import load_versions
+
+        rules_versions = load_versions
+
+        if self.id in rules_versions:
+            version_info = rules_versions[self.id]
+            version = version_info['version']
+            return version + 1 if self.get_hash() != version_info['sha256'] else version
+        else:
+            return 1
+
+    def get_payload(self, include_version=False, randomize_id=False, embed_metadata=False, target_version=None):
```

Comment:
personally, I like `replace_id` or `generate_id` or something like that. randomize just seems a little odd to me for some reason

Reply:
That's right - now I remember you saying this in another PR. Will update, thanks